src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2982,6 +3092,13 @@ StartPlacementExecutionOnSession(TaskPlacementExecution *placementExecution,
 }
 
 
+static void
+LogSessionId(uint64 sessionId)
+{
+	ereport(DEBUG4, (errmsg("Command will be sent on session id: %ld", sessionId)));
```

Comment:
It would be nicer if this was part of the same line. Maybe we should use "connection ID" (stored in MultiConnection) instead of a session ID everywhere?

Reply:
You mean we should create connectionID for MultiConnection? Because as far as I see, it doesnt a connection does not have an id.

Reply:
yes, could be useful to add some static counter to StartConnectionEstablishment

Reply:
I think in that case we can just add this to LogRemoteCommand:

```
LogRemoteCommand(MultiConnection *connection, const char *command)
{
	if (!LogRemoteCommands)
	{
		return;
	}

	ereport(LOG, (errmsg("issuing %s", ApplyLogRedaction(command)),
				  errdetail("on server %s@%s:%d connectionId: %ld", connection->user, connection->hostname,
							connection->port, connection->connectionId)));
}

```

Reply:
yes, that what I was thinking